morpheus/models/dfencoder/dataloader.py

```diff
@@ -545,65 +499,34 @@ def _preprocess(self, df, batch_index):
 
-    @staticmethod
-    def get_train_dataset(model, df):
-        """A helper function to get a train dataset with the provided parameters.
+    @property
+    def preprocess_fn(self):
+        return self._preprocess_fn
 
-        Parameters
-        ----------
-        model : AutoEncoder
-            The autoencoder model used to get relevant params and the preprocessing func.
-        df : pandas.DataFrame
-            Input dataframe used for the dataset.
+    @preprocess_fn.setter
+    def preprocess_fn(self, value):
+        self._preprocess_fn = value
 
-        Returns
-        -------
-        DatasetFromDataframe
-            Training Dataset set up to load from the dataframe.
-        """
-        dataset = DatasetFromDataframe(
-            df=df,
-            batch_size=model.batch_size,
-            preprocess_fn=model.preprocess_train_data,
-            shuffle_rows_in_batch=True,
-            shuffle_batch_indices=True,
-        )
-        return dataset
+    @property
+    def batch_size(self):
+        return self._batch_size
 
-    @staticmethod
-    def get_validation_dataset(model, df):
-        """A helper function to get a validation dataset with the provided parameters.
+    @batch_size.setter
+    def batch_size(self, value):
+        self._batch_size = value
 
-        Parameters
-        ----------
-        model : AutoEncoder
-            The autoencoder model used to get relevant params and the preprocessing func.
-        df : pandas.DataFrame
-            Input dataframe used for the dataset.
+    @property
+    def shuffle_rows_in_batch(self):
+        return self._shuffle_rows_in_batch
 
-        Returns
-        -------
-        DatasetFromDataframe
-            Validation Dataset set up to load from the dataframe.
-        """
-        dataset = DatasetFromDataframe(
-            df=df,
-            batch_size=model.eval_batch_size,
-            preprocess_fn=model.preprocess_validation_data,
-            shuffle_rows_in_batch=False,
-            shuffle_batch_indices=False,
-        )
-        return dataset
+    @shuffle_rows_in_batch.setter
+    def shuffle_rows_in_batch(self, value):
+        self._shuffle_rows_in_batch = value
 
-    def convert_to_validation(self, model):
-        """Converts the dataset to validation mode by resetting instance variables.
+    @property
+    def shuffle_batch_indices(self):
+        return self._shuffle_batch_indices
 
-        Parameters
-        ----------
-         model : AutoEncoder
-            The autoencoder model used to get relevant params and the preprocessing func.
-        """
-        self._preprocess_fn = model.preprocess_validation_data
-        self._batch_size = model.eval_batch_size
-        self._shuffle_rows_in_batch = False
-        self._shuffle_batch_indices = False
+    @shuffle_batch_indices.setter
+    def shuffle_batch_indices(self, value):
+        self._shuffle_batch_indices = value
```

Comment:
These are the same between the two dataset classes in this file. Should they be abstracted to a common base class?

Reply:
Hsin and I talked about it; since they're specific to DFP, and the classes themselves are fairly distinct in terms of function, I'm kind of indifferent. I also expect this will change or go away entirely when we transition to ControlMessages with managed payloads.

Reply:
Fair enough.